From c7636fb343c848000300a9429410b74d17a4d54d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=96mer=20Fad=C4=B1l=20Usta?= <omerusta@gmail.com>
Date: Tue, 8 Jun 2021 05:13:23 +0000
Subject: [PATCH] Partial update of Commit 62f51070 to make it compileable
 again

Cmake to 3.16
Fix Fontconfig's FOUND variable
Fix Fontconfig's include_dirs variable
Use Correct Target to link FontConfig and Freetype
---
 CMakeLists.txt           | 14 +++++---------
 libs/text/CMakeLists.txt |  4 ++--
 2 files changed, 7 insertions(+), 11 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1ce210ebe01..71272ffef2d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.16)
 
 project(calligra)
 
@@ -168,6 +168,8 @@ calligra_set_productset(${PRODUCTSET})
 ## Look for ECM, Qt, KF5 ##
 ###########################
 ##########################
+set(REQUIRED_KF5_VERSION "5.7.0")
+set(REQUIRED_QT_VERSION "5.3.0")
 
 find_package(ECM 5.19 REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
@@ -199,9 +201,6 @@ include(MacroOptionalFindPackage)
 include(MacroEnsureVersion)
 include(MacroDesktopToJson)
 
-
-set(REQUIRED_KF5_VERSION "5.7.0")
-
 find_package(KF5 ${REQUIRED_KF5_VERSION} REQUIRED
     COMPONENTS
     Archive
@@ -251,9 +250,6 @@ if(${KF5_VERSION} VERSION_LESS "5.16.0")
     set(CALLIGRA_OLD_PLUGIN_METADATA TRUE)
 endif()
 
-
-set(REQUIRED_QT_VERSION "5.3.0")
-
 find_package(Qt5 ${REQUIRED_QT_VERSION} REQUIRED
     COMPONENTS
     Core
@@ -641,8 +637,8 @@ if(NOT WIN32 AND NOT APPLE)
 	)
 endif()
 
-if(NOT FONTCONFIG_FOUND OR NOT FREETYPE_FOUND)
-    set(FONTCONFIG_INCLUDE_DIR "")
+if(NOT Fontconfig_FOUND OR NOT FREETYPE_FOUND)
+    set(Fontconfig_INCLUDE_DIRS "")
     set(FREETYPE_INCLUDE_DIRS "")
 else()
     add_definitions( -DSHOULD_BUILD_FONT_CONVERSION )
diff --git a/libs/text/CMakeLists.txt b/libs/text/CMakeLists.txt
index 16d7e066fa2..429afe79704 100644
--- a/libs/text/CMakeLists.txt
+++ b/libs/text/CMakeLists.txt
@@ -152,11 +152,11 @@ if( SHOULD_BUILD_FEATURE_RDF )
 endif()
 
 if( FONTCONFIG_FOUND )
-    target_link_libraries(kotext PRIVATE ${FONTCONFIG_LIBRARIES})
+    target_link_libraries(kotext PRIVATE Fontconfig::Fontconfig)
 endif()
 
 if( FREETYPE_FOUND )
-    target_link_libraries(kotext PRIVATE ${FREETYPE_LIBRARIES})
+    target_link_libraries(kotext PRIVATE Freetype::Freetype)
 endif()
 
 
-- 
GitLab

